src/Microsoft.AspNet.OData.Shared/Formatter/ODataOutputFormatterHelper.cs

```diff
@@ -349,5 +348,52 @@ private static bool IsOperationPath(ODataPath path)
 
             return false;
         }
+
+        private static Microsoft.OData.UriParser.ODataPath ConvertPath(ODataPath path)
+        {
+            if (path == null)
+            {
+                return null;
+            }
+
+            if (IsOperationPath(path))
+            {
+                var lastSegment = path.Segments.Last();
+                OperationSegment operation = lastSegment as OperationSegment;
+                if (operation != null && operation.EntitySet != null)
+                {
+                    return GeneratePath(operation.EntitySet);
+                }
+
+                OperationImportSegment operationImport = lastSegment as OperationImportSegment;
+                if (operationImport != null && operationImport.EntitySet != null)
+                {
+                    return GeneratePath(operationImport.EntitySet);
+                }
+
+                return null;
+            }
+
+            return path.Path;
+        }
+
+        private static Microsoft.OData.UriParser.ODataPath GeneratePath(IEdmEntitySetBase entitySetBase)
```

Comment:
Minor: Can move it to a Factory implementation.

Reply:
This is only used in this class once. So, it's better to keep it private in this helper class.

---
In reply to: [384850926](https://github.com/OData/WebApi/pull/2060#discussion_r384850926) [](ancestors = 384850926)